### PR TITLE
Add locutil.rs

### PR DESCRIPTION
**Pull request by darkHarry**
TODO(@darkHarry ):

- [ ] import lineBreakG function from whitespace module
- [ ] define types for all function params and struct members
- [ ] add getLineInfo function
- [ ] check for p.sourceFile's default value (currently used is `()` )

### CODE REVIEW

File: src/locutil.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// TODO import lineBreakG fn from whitespace
```
Comment (ryzokuken) [inline src/locutil.rs:1]: I guess a nice TODO format would be:

```
TODO(Harry): do something
```

File: src/locutil.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// TODO define types for all fn parameters and structs
+// TODO add getLineInfo fn
+
+// These are used when `options.locations` is on, for the
+// `startLoc` and `endLoc` properties.
+
+mod locutil {
```
Comment (ryzokuken) [inline src/locutil.rs:8]: As discussed, you will need to remove this.

File: src/locutil.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    pub struct Position {
+        line: u64,
```
Comment (ryzokuken) [inline src/locutil.rs:10]: would we ever realistically need numbers this big?

File: src/locutil.rs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        column: u64,
+    }
+
+    impl Position {
+        pub fn new(line: u64, column: u64) -> Position {
+            Position{ line: line, column: column }
+        }
+
+        pub fn offset(&self, n) -> Position {
```
Comment (ryzokuken) [inline src/locutil.rs:19]: did you run this code? This is incorrect, `n` has no type specified.

File: src/locutil.rs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            Position{ line: self.line, column: self.column + n}
+        }
+    }
+
+    pub struct SourceLocation {
```
Comment (ryzokuken) [inline src/locutil.rs:24]: no types.

File: src/locutil.rs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        start,
+        end,
+        source,
+    }
+
+    impl SourceLocation {
+        pub fn new(p, start, end) -> SourceLocation {
+            if p.sourceFile != () { // TODO what should the empty value be ?
```
Comment (ryzokuken) [inline src/locutil.rs:32]: Depends on the type of `sourceFile`.

**Review by ryzokuken — CHANGES_REQUESTED** [review]
A few comments.

File: src/locutil.rs (hunk 1 above)
Comment (darkHarry) [reply src/locutil.rs:1]: ig it's pretty obvious that i have to do it.

Comment (ryzokuken) [reply src/locutil.rs:1]: I mean, overtime there will be too many TODOs in the codebase, it'd be helpful to see everyone's TODOs separately.

File: src/locutil.rs (hunk 2 above)
Comment (darkHarry) [reply src/locutil.rs:8]: yes.

File: src/locutil.rs (hunk 3 above)
Comment (darkHarry) [reply src/locutil.rs:10]: No assumptions! Column won't be but what can we say about lines?

File: src/locutil.rs (hunk 4 above)
Comment (darkHarry) [reply src/locutil.rs:19]: Read the TODO.

**Comment by darkHarry** [comment]
Don't need it anymore.

**Comment by ryzokuken** [comment]
Dude, I told you I'll delete my code that conflicts with yours.

**Comment by darkHarry** [comment]
I'll work on yours. Are you gonna merge yours now or later?

**Comment by ryzokuken** [comment]
@darkHarry depends. If you think it's good to go, let's merge and push things one function at a time after that.

**Pull request closed without merging**